openedx_proversity_reports/tasks.py

```diff
@@ -69,3 +68,25 @@ def generate_last_page_accessed_report(courses):
         report_data['exit_count_data'] = get_exit_count_data(last_page_data, courses)
 
     return report_data
+
+
+@task(default_retry_delay=5, max_retries=5)  # pylint: disable=not-callable
+def generate_time_spent_report(courses):
+    """
+    Return the time spent data for the given courses.
+
+    Args:
+        courses: Course ids list.
+    Returns:
+        Dict with 'time_spent_data' containing time spent report data.
+    """
+    report_data = {
+        'time_spent_data': {},
+    }
+
+    time_spent_report_data = get_time_spent_report_data(courses)
+
+    if time_spent_report_data:
+        report_data['time_spent_data'] = time_spent_report_data
+
+    return report_data
```

Comment:
you can remove lines 83, 84,85, 89 and 90 and merge all in one line 
```
return {'time_spent_data': time_spent_report_data if time_spent_report_data else {}}
```

Reply:
Great catch. @andrey-canon